File: API/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{    public class Order
    {
        public int Id { get; set; }
        public virtual ICollection<OrderDish> OrderItems { get; set; } = new List<OrderDish>();
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public decimal TotalAmount { get; set; } = 0;
        public string PaymentMethod { get; set; } = string.Empty;
        public string Status { get; set; } = "Pending"; // Change to ENUM Later.

        public int UserId { get; set; }
        public virtual User User { get; set; } = null!; // Navigation property to User (Customer)
        public int RestaurantId { get; set; }
        public virtual Restaurant Restaurant { get; set; } = null!;  // Navigation property

        public int? EmployeeId { get; set; } // Nullable to allow order creation without delivery person
        public virtual User? Employee { get; set; } // Fixed the nullable reference

        // Delivery address information
        public int? DeliveryAddressId { get; set; }
        public virtual Address? DeliveryAddress { get; set; }
        public string? DeliveryInstructions { get; set; } = string.Empty;

        [NotMapped]
        public decimal CalculatedTotalAmount => OrderItems.Sum(item => item.Quantity * item.UnitPrice);
    }
}

File: careem_clone_mobile/lib/models/order.dart
import 'dish.dart';

class Order {
  final int id;
  final int userId;
  final int restaurantId;
  final String status;
  final double totalAmount;
  final DateTime createdAt;
  final List<OrderItem>? items;
  
  Order({
    required this.id,
    required this.userId,
    required this.restaurantId,
    required this.status,
    required this.totalAmount,
    required this.createdAt,
    this.items,
  });
  
  factory Order.fromJson(Map<String, dynamic> json) {
    List<OrderItem>? orderItems;
    if (json['items'] != null) {
      orderItems = (json['items'] as List)
          .map((item) => OrderItem.fromJson(item))
          .toList();
    }
    
    return Order(
      id: json['id'],
      userId: json['userId'],
      restaurantId: json['restaurantId'],
      status: json['status'],
      totalAmount: json['totalAmount']?.toDouble() ?? 0.0,
      createdAt: DateTime.parse(json['createdAt']),
      items: orderItems,
    );
  }
}

class OrderItem {
  final int dishId;
  final int quantity;
  final double unitPrice;
  final Dish? dish;
  
  OrderItem({
    required this.dishId,
    required this.quantity,
    required this.unitPrice,
    this.dish,
  });
  
  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      dishId: json['dishId'],
      quantity: json['quantity'],
      unitPrice: json['unitPrice']?.toDouble() ?? 0.0,
      dish: json['dish'] != null ? Dish.fromJson(json['dish']) : null,
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'dishId': dishId,
      'quantity': quantity,
    };
  }
}
